Make sure the JAXBContext is initialized at build time when building native

gnodet opened the pull request:
Fixes #2205

gnodet committed:
Make sure the JAXBContext is initialized at build time when running in native mode, fixes #2205

gnodet commented:
This also relates to #2417

gsmet commented on extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java:
```diff
@@ -134,8 +134,8 @@ void process(BuildProducer<SubstrateSystemPropertyBuildItem> substrateProps,
-        runtimeClasses
-                .produce(new RuntimeInitializedClassBuildItem("com.sun.xml.internal.bind.v2.runtime.reflect.opt.Injector"));
+        //        runtimeClasses
+        //                .produce(new RuntimeInitializedClassBuildItem("com.sun.xml.internal.bind.v2.runtime.reflect.opt.Injector"));
```
Mmmmh?

gnodet replied:
My understanding is that the JAXB `com.sun.xml.internal.bind.v2.runtime.reflect.opt.Injector` class is only used from `com.sun.xml.internal.bind.v2.runtime.reflect.opt.AccessorInjector#prepare` method, which is substituted to an empty method (see https://github.com/quarkusio/quarkus/blob/master/extensions/jaxb/runtime/src/main/java/io/quarkus/jaxb/runtime/graal/JAXBSubstitutions.java#L83).  The result is that there is no reference to this class anymore and GraalVM throws an exception when building compiling to native.
I suppose this may have been caused by a change in Graal.

gnodet replied:
@gsmet I added another commit to remove the commented code

gsmet replied:
@gnodet cool thanks. Sorry to be picky but could you amend the commit message with your explanation. That would be useful for history if one day we ask ourselves why we removed it.

Once it's done, I'll merge.

gnodet replied:
@gsmet done

oscerd replied:
this one is really important to make the camel extension works.

gsmet reviewed (changes requested):
Wondering why you commented the `RuntimeInitializedClassBuildItem` addition? Should it be removed? Is it an error?

oscerd commented:
Can we get this one merged, This would fix the integration tests for camel-*

gsmet commented:
@oscerd as mentioned above, there is some code commented out and I don't know if it's a mistake or code we should remove.

I won't merge it until I get a proper answer.

oscerd commented:
Yes, just noticed that sorry @gsmet

oscerd commented:
I'm testing the effect of this one btw

oscerd commented:
@gsmet I can confirm this fix the integration tests, lets wait for @gnodet and sorry for the noise

oscerd commented:
And btw it would be better to completely remove that stuff, without comment as you said.

gsmet commented:
@oscerd it might make sense for Camel but JAXB is consumed by other extensions and I would like to understand the rationale of this change before merging it.

oscerd commented:
I tested different integration tests even not camel related. Just for completeness :-)

gnodet committed:
Remove the build item completely

My understanding is that the JAXB com.sun.xml.internal.bind.v2.runtime.reflect.opt.Injector class is only used from com.sun.xml.internal.bind.v2.runtime.reflect.opt.AccessorInjector#prepare method, which is substituted to an empty method (see https://github.com/quarkusio/quarkus/blob/master/extensions/jaxb/runtime/src/main/java/io/quarkus/jaxb/runtime/graal/JAXBSubstitutions.java#L83). The result is that there is no reference to this class anymore and GraalVM throws an exception when building compiling to native.
I suppose this may have been caused by a change in Graal.
